### DISCUSSION TITLE
esphome alarm panel without esp32 s3 box 3

### DISCUSSION THREAD

**Discussion by youkorr** [BigBobbas/ESP32-S3-Box3-Custom-ESPHome#75] [General]
Hello everyone I hope you are well
I am trying to introduce a page for an alarm keyboard but everything I do does not work or it does not appear as I am not good at yaml I would have help if you can thank you

**Comment by youkorr** [comment]
I am with a frigate app on home assistant on I use this keyboard to activate my alarm
![Capture d’écran 2024-08-11 154643](https://github.com/user-attachments/assets/987512ab-14f3-40bd-a454-39ee5c2f0eb5)

**Comment by BigBobbas** [comment]
I don't think that there is support for this in esphome. As I have looked into this previously for use with Alarmo. However I will have a look further into this when I am near a computer as I am on mobile at the moment.
Thanks
Rob

**Comment by Kelerion-UK** [comment]
I don't use HA for alarms... but if there is already an integration I am sure there must be service/actions for it? I'm sure you'd only have to call those?

**Comment by youkorr** [comment]
ok thanks I will do my research and I will let you know

**Comment by BigBobbas** [comment]
That's what I have done with the s3 watch for setting and disarming the alarm by using a service call and not using a code.

**Comment by BigBobbas** [comment]
You could emulate a keypad by displaying an image of one and mapping binary sensors to each keypad button and then have a switch template for each number. And a script to check which 'switches' are on and then trigger service call to arm or disarm... bit hacky but would kind of work lol

**Comment by youkorr** [comment]
I see but I think it is possible to introduce it on your custom firmware

**Comment by youkorr** [comment]
create a home assistant alarm deporter keyboard

**Reply by BigBobbas** [reply]
Link ? Info ? Needs to be available in esphome https://esphome.io

**Comment by youkorr** [comment]
great thanksgreat thanks and have a good holiday

**Comment by youkorr** [comment]
Hello, I am facing this problem that I cannot resolve. 
Component not found: input_text.
 pin_input: name: PIN Input max_length: 4

**Comment by youkorr** [comment]
`esphome:
  name: alarm_control_panel

esp32:
  board: esp32dev
  framework:
    type: esp-idf

wifi:
  ssid: "Votre_SSID"
  password: "Votre_Mot_de_Passe"

  ap:
    ssid: "AlarmControlPanel"
    password: "SecurePassword"

captive_portal:

# Panneau de contrôle d'alarme avec code PIN
display:
  - platform: ssd1306_i2c
    model: "SSD1306 128x64"
    rotation: 0
    address: 0x3C
    update_interval: 1s
    lambda: |-
      it.print(0, 0, id(font), "Enter PIN:");
      it.print(0, 20, id(font), id(pin_input).c_str());

text_sensor:
  - platform: template
    name: "Alarm Panel Status"
    id: alarm_status
    icon: "mdi:shield-home"

input_text:
  pin_input:
    name: "PIN Input"
    max_length: 4

binary_sensor:
  - platform: gpio
    pin: 0
    name: "Alarm Arm"
    on_press:
      then:
        - lambda: |-
            if (id(pin_input).state == "1234") {
              id(alarm_status).publish_state("Armed");
            } else {
              id(alarm_status).publish_state("Invalid PIN");
            }

  - platform: gpio
    pin: 1
    name: "Alarm Disarm"
    on_press:
      then:
        - lambda: |-
            if (id(pin_input).state == "1234") {
              id(alarm_status).publish_state("Disarmed");
            } else {
              id(alarm_status).publish_state("Invalid PIN");
            }

font:
  - file: "fonts/OpenSans-Regular.ttf"
    id: font
    size: 20

mqtt:
  broker: "192.168.1.100"  # Adresse IP de votre broker MQTT (Home Assistant)
  username: "mqtt_user"
  password: "mqtt_password"

api:

ota:
`

**Comment by youkorr** [comment]
or without frigate

`# Page d'accueil pour le contrôle de l'alarme
display:
  - platform: st7789v
    pages:
      - id: alarm_page
        lambda: |-
          it.fill(COLOR_BLACK);
          it.print(10, 10, id(my_font), "Alarm Control");

# Boutons pour armer/désarmer l'alarme
binary_sensor:
  - platform: gpio
    pin: GPIO_NUM_X
    name: "Armer Alarme"
    on_press:
      then:
        - homeassistant.service:
            service: alarm_control_panel.alarm_arm_away
            data:
              entity_id: alarm_control_panel.your_alarm_entity
  - platform: gpio
    pin: GPIO_NUM_Y
    name: "Désarmer Alarme"
    on_press:
      then:
        - homeassistant.service:
            service: alarm_control_panel.alarm_disarm
            data:
              entity_id: alarm_control_panel.your_alarm_entity
`

**Reply by BigBobbas** [reply]
try using a text component https://esphome.io/components/text/index.html

**Comment by youkorr** [comment]
I know I'm not very far from integrating it with your firmware, thank you.

**Reply by youkorr** [reply]
J'ai reussi a mettre le clavier il faut que j'arrive a le reduire
![CLAVIER](https://github.com/user-attachments/assets/a0876e40-ebfd-40ba-8f80-a281d7cdb4b7)

**Reply by BigBobbas** [reply]
nice work - hope that you can figure it out 👍

**Comment by youkorr** [comment]
`          it.fill(id(black));
          it.print(10, 10, id(font1), "Enter PIN:");
          for (int i = 0; i < 3; i++) {
            for (int j = 0; j < 3; j++) {
              int number = i * 3 + j + 1;
              it.rectangle(20 + j*60, 50 + i*60, 50, 50, light_blue);
              char num_str[2];  // Tableau pour stocker le numéro sous forme 
              snprintf(num_str, sizeof(num_str), "%d", number);  // Convertit le nombre en chaîne
              it.print(45 + j*60, 75 + i*60, id(font2), TextAlign::CENTER, num_str);;
            }
          }
          it.rectangle(10, 230, 120, 50, blue);
          it.print(80, 255, id(font2), TextAlign::CENTER, "0");
          it.rectangle(140, 230, 120, 50, red);
          it.print(200, 255, id(font2), TextAlign::CENTER, "OK");`

**Comment by youkorr** [comment]
![ALARM](https://github.com/user-attachments/assets/294e1117-d766-455d-b23a-7ed02f6c1e36)

**Reply by youkorr** [reply]
Hello I need help to insert rectangles on the right
![CLAVIER](https://github.com/user-attachments/assets/8a10d884-1682-4ef9-846d-2fc080cdd4b9)

**Reply by BigBobbas** [reply]
try this ,  the first number = how far accross from the left to start , the second is how far from the top to start. the third is how long the rectangle is and the fourth is how high to draw it.
it.rectangle(220 , 100 , 60 , 40 , blue);
it.rectangle(220 , 150 , 60 , 40 , blue);

**Reply by youkorr** [reply]
Great thank you very much you helped me a lot I will miss that the touchscreen from 1 to 9 and armed and disarmed is functional but I risk having a GPIO problem for the touchscreen

**Comment by youkorr** [comment]
![clavier alarmo](https://github.com/user-attachments/assets/b341b608-6fb8-49bd-9c36-3c92e19da476)

**Comment by youkorr** [comment]
Good evening despite several attempts and searches the keys do not work I do not know where it comes from I provide this code which does not work hoping that someone can resolve it thank you

global
`  - id: pin_code
    type: std::string
    initial_value: ""  `
`script:
  - id: press_digit_1
    then:
      - lambda: |-
          id(pin_code) += "1";

  - id: press_digit_2
    then:
      - lambda: |-
          id(pin_code) += "2";

  - id: press_digit_3
    then:
      - lambda: |-
          id(pin_code) += "3";

  - id: validate_pin_action
    then:
      - lambda: |-
          if (id(pin_code).length() == 4) {
            if (id(pin_code) == "1234") {  # Remplacez "1234" par votre code PIN
              ESP_LOGI("validate_pin", "PIN correct");
              // Envoyer une commande pour désarmer l'alarme
              id(pin_code).clear();
            } else {
              ESP_LOGI("validate_pin", "PIN incorrect");
              id(pin_code).clear();
            }
          }
          
globals:
  - id: pin_code
    type: std::string
    initial_value: ""`

`binary_sensor:
  - platform: touchscreen
    name: "Validate PIN"
    x_min: 10
    x_max: 180
    y_min: 220
    y_max: 260
    on_press:
      then:
        - script.execute: validate_pin_action
        - 
` - id: other_set_page ##### other settings page #####
        lambda: |-
          it.fill(id(black));
          it.print(10, 10, id(font1), "Alarm Control");
          for (int i = 0; i < 3; i++) {
            for (int j = 0; j < 3; j++) {
              int number = i * 3 + j + 1;
              it.rectangle(15 + j*60, 50 + i*60, 50, 50, light_blue);
              char num_str[2];  // Tableau pour stocker le numéro sous forme de chaîne
              snprintf(num_str, sizeof(num_str), "%d", number);  // Convertit le nombre en chaîne
              it.print(45 + j*60, 75 + i*60, id(font2), TextAlign::CENTER, num_str);;
            }
          }
          it.filled_rectangle(198, 50, 108, 50, blue);
          it.print(255, 75, id(font3), white, TextAlign::CENTER, "armed");
          it.filled_rectangle(198, 170, 108, 50, red);
          it.print(250, 195, id(font3), white, TextAlign::CENTER, "disarmed");`

  - platform: touchscreen
    name: "Button 1"
    x_min: 10
    x_max: 60
    y_min: 10
    y_max: 60
    on_press:
      then:
        - lambda: |-
            ESP_LOGI("touch", "Button 1 pressed!");
            id(pin_code) += "1";

**Reply by BigBobbas** [reply]
if you watch the device logs, does anything show when you press the key digits?

**Reply by youkorr** [reply]
ok i'll see that

**Reply by youkorr** [reply]
when I touch a number on the keyboard it tells me this?? I must have made a mistake


[19:11:37][I][touch:2907]: Button 1 pressed!

**Reply by BigBobbas** [reply]
in your binary sensors you also need to add `page_id: other_set_page' too to stop button presses on the keypad page translating to other pages.

Have you created a binary_sensor for each digit ?

**Reply by youkorr** [reply]
for now I did it for 3 digits to do tests,for now I did it for 3 digits to do tests
Or I can put add `page_id: other_set_page

  - platform: touchscreen
    name: "Button 1"
    x_min: 10
    x_max: 60
    y_min: 10
    y_max: 60
    on_press:
      then:
        - lambda: |-
            ESP_LOGI("touch", "Button 1 pressed!");
            id(pin_code) += "1";

**Reply by youkorr** [reply]
ok I hadn't seen it I had forgotten it

`  - platform: touchscreen
    page_id: other_set_page 
    name: "Button 1"
    x_min: 10
    x_max: 60
    y_min: 10
    y_max: 60
    on_press:
      then:
        - lambda: |-
            ESP_LOGI("touch", "Button 1 pressed!");
            id(pin_code) += "1";`

**Comment by BigBobbas** [comment]
whne you paste in code if you put 3 back tics followed by `yaml` before your code and 3 back ticks after the last line, then the code will be shown in a readable format eg:
![image](https://github.com/user-attachments/assets/ae049e05-059f-42ae-bade-c2b21db83b02)

will look like the code below, which makes it easier to read for myself and others.

```yaml
platform: touchscreen
name: "Button 1"
x_min: 10
x_max: 60
y_min: 10
y_max: 60
on_press:
then:
- lambda: |-
ESP_LOGI("touch", "Button 1 pressed!");
id(pin_code) += "1";
```

**Reply by youkorr** [reply]
ok I didn't know that thanks for pointing it out to me

**Reply by youkorr** [reply]
```yaml
  - platform: touchscreen
    page_id: other_set_page 
    name: "Button 1"
    x_min: 10
    x_max: 60
    y_min: 10
    y_max: 60
    on_press:
      then:
        - lambda: |-
            ESP_LOGI("touch", "Button 1 pressed!");
            id(pin_code) += "1";
```

**Reply by youkorr** [reply]
good evening for the keys it works but the armed and disarmed part I have a compilation error despite the fact that I did clean, it would be missing that I declare disarmed in section adf_pipeline: ??
There I am stuck

error: 'desarmed' was not declared in this scope

**Reply by youkorr** [reply]
note: suggested alternative: 'rename'
       model: es8311

**Reply by youkorr** [reply]
Good for the keys works perfectly I added the OK key to validate pin action
![clavier modi](https://github.com/user-attachments/assets/c5f56cbf-0132-4c0b-8334-5d9c80e5529f)

**Reply by BigBobbas** [reply]
> good evening for the keys it works but the armed and disarmed part I have a compilation error despite the fact that I did clean, it would be missing that I declare disarmed in section adf_pipeline: ?? There I am stuck
> 
> error: 'desarmed' was not declared in this scope

the adf_pipeline is for audio on the device, so not relevant to this

**Reply by youkorr** [reply]
Thanks but I will find another solution to get around this problem.

**Reply by BigBobbas** [reply]
if you paste the code and the error. i can take a look 👍

**Reply by youkorr** [reply]
The way I found is to recover alarm_control_panel.alarmo from homeassistant as soon as I get stuck I will tell you thank you for your help

**Comment by youkorr** [comment]
![Capture d’écran 2024-08-16 031723](https://github.com/user-attachments/assets/1632db1f-ce64-4a1e-92ca-0eb5f0bbda06)

**Reply by youkorr** [reply]
the buttons I made it appear to check if it works

**Comment by BigBobbas** [comment]
it may be easier for you if you join our discord server, then you can have conversations in 'real time' as the posts in here become fragment and it can be hard to follow https://discord.gg/BfEc3pXe

**Comment by youkorr** [comment]
I thank Rob for his patience the alarm control panel is functional if you want the code it will be a pleasure
